editor/scene_tree_editor.cpp

```diff
@@ -270,15 +270,37 @@ bool SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 			item->add_button(0, get_icon("NodeWarning", "EditorIcons"), BUTTON_WARNING, false, TTR("Node configuration warning:") + "\n" + p_node->get_configuration_warning());
 		}
 
-		bool has_connections = p_node->has_persistent_signal_connections();
-		bool has_groups = p_node->has_persistent_groups();
-
-		if (has_connections && has_groups) {
-			item->add_button(0, get_icon("SignalsAndGroups", "EditorIcons"), BUTTON_SIGNALS, false, TTR("Node has connection(s) and group(s).\nClick to show signals dock."));
-		} else if (has_connections) {
-			item->add_button(0, get_icon("Signals", "EditorIcons"), BUTTON_SIGNALS, false, TTR("Node has connections.\nClick to show signals dock."));
-		} else if (has_groups) {
-			item->add_button(0, get_icon("Groups", "EditorIcons"), BUTTON_GROUPS, false, TTR("Node is in group(s).\nClick to show groups dock."));
+		int num_connections = p_node->get_persistent_signal_connection_count();
+		int num_groups = p_node->get_persistent_group_count();
+
+		if (num_connections >= 1 && num_groups >= 1) {
+			item->add_button(
+					0,
+					get_icon("SignalsAndGroups", "EditorIcons"),
+					BUTTON_SIGNALS,
+					false,
+					vformat(
+							TTR("Node has %s and %s.\nClick to show signals dock."),
+							vformat(num_connections == 1 ? TTR("%d connection") : TTR("%d connections"), num_connections),
```

Comment:
Note that this is not going to work for languages with plural forms different from English (1 = singular, everything else = plural). For example, Arabic has six forms, Japanese doesn't really have a plural at all: https://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html.

The gettext format has [a way to to handle this](https://www.gnu.org/software/gettext/manual/html_node/Translating-plural-forms.html) but I don't think Godot's translation system has support for it at the moment.

Reply:
I think it's not worth it to handle plurals manually this way as nested substitutions, it's pretty confusing to translate in some languages IMO.

If we had proper plurals support from gettext, we can revise this, but until then I'd stick to a clearer "Node has %d connection(s) and %d group(s).".